From: Jiri Slaby <jslaby@suse.cz>
Subject: sound: ratelimit sound debug prints
Patch-mainline: not yet

It's printed many times each second. This is only a workaround. Has to
be fixed properly.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 sound/usb/urb.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/sound/usb/urb.c
+++ b/sound/usb/urb.c
@@ -17,6 +17,7 @@
 
 #include <linux/gfp.h>
 #include <linux/init.h>
+#include <linux/ratelimit.h>
 #include <linux/usb.h>
 #include <linux/usb/audio.h>
 
@@ -458,7 +459,7 @@ static int retire_capture_urb(struct snd
 
 	for (i = 0; i < urb->number_of_packets; i++) {
 		cp = (unsigned char *)urb->transfer_buffer + urb->iso_frame_desc[i].offset;
-		if (urb->iso_frame_desc[i].status) {
+		if (urb->iso_frame_desc[i].status && printk_ratelimit()) {
 			snd_printd(KERN_ERR "frame %d active: %d\n", i, urb->iso_frame_desc[i].status);
 			// continue;
 		}
